javaee8/security-1.0/src/main/java/org/wildfly/swarm/ts/javaee8/security/customformauth/CustomIdentityStore.java

```diff
@@ -0,0 +1,45 @@
+package org.wildfly.swarm.ts.javaee8.security.customformauth;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.security.enterprise.credential.UsernamePasswordCredential;
+import javax.security.enterprise.identitystore.CredentialValidationResult;
+import javax.security.enterprise.identitystore.IdentityStore;
+
+@ApplicationScoped
+public class CustomIdentityStore implements IdentityStore {
+    private static Map<String, String> emailsToPasswords;
+
+    private static Map<String, String> emailsToRoles;
+
+    static {
+        emailsToPasswords = new HashMap<String, String>() {
+            {
```

Comment:
Using instance initializers to emulate map literals is particularly despicable. I think we have Guava in our dependencies, use their `ImmutableMap`.